src/python_testing/TC_IDM_4_2.py

```diff
@@ -0,0 +1,582 @@
+
+        return acl_list
+
+    async def add_ace_to_dut_acl(self, ctrl, ace):
+        dut_acl_original = await self.get_dut_acl()
+        dut_acl = copy.deepcopy(dut_acl_original)
+        dut_acl.append(ace)
+        await self.write_acl(ctrl=ctrl, acl=dut_acl)
+
+    @staticmethod
+    def is_valid_uint32_value(var):
+        return isinstance(var, int) and 0 <= var <= 0xFFFFFFFF
+
+    @staticmethod
+    def is_valid_uint16_value(var):
+        return isinstance(var, int) and 0 <= var <= 0xFFFF
+
+    @async_test_body
+    async def test_TC_IDM_4_2(self):
+
+        # Test setup
+        cluster_rev_attr = Clusters.BasicInformation.Attributes.ClusterRevision
+        cluster_rev_attr_typed_path = self.get_typed_attribute_path(cluster_rev_attr)
+        node_label_attr = Clusters.BasicInformation.Attributes.NodeLabel
+        node_label_attr_path = [(0, node_label_attr)]
+        node_label_attr_typed_path = self.get_typed_attribute_path(node_label_attr)
+        SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT_SEC = 0
+        INVALID_ACTION_ERROR_CODE = 0x580
+
+        # Controller 1 setup
+        # Subscriber/client with admin access to the DUT
+        # Will write ACL for controller 2 and validate success/error codes
+        CR1: ChipDeviceController = self.default_controller
+
+        # Controller 2 setup
+        # Subscriber/client with limited access to the DUT
+        # Will validate error status codes
+        fabric_admin = self.certificate_authority_manager.activeCaList[0].adminList[0]
+        CR2_nodeid = self.matter_test_config.controller_node_id + 1
+        CR2: ChipDeviceController = fabric_admin.NewController(
+            nodeId=CR2_nodeid,
+            paaTrustStorePath=str(self.matter_test_config.paa_trust_store_path),
+        )
+
+        # Read ServerList attribute
+        self.print_step("0a", "CR1 reads the Descriptor cluster ServerList attribute from EP0")
+        ep0_servers = await self.get_descriptor_server_list(CR1)
+
+        # Check if ep0_servers contains the ICD Management cluster ID (0x0046)
+        if Clusters.IcdManagement.id in ep0_servers:
+            # Read the IdleModeDuration attribute value from the DUT
+            logging.info(
+                "CR1 reads from the DUT the IdleModeDuration attribute and sets SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT_SEC = IdleModeDuration")
+
+            idleModeDuration = await self.get_idle_mode_duration_sec(CR1)
+
+            SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT_SEC = idleModeDuration
+        else:
+            # Defaulting SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT_SEC to 60 minutes
+            SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT_SEC = 60 * 60
+
+        logging.info(
+            f"Set SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT_SEC to {SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT_SEC} seconds")
+
+        '''
+        ##########
+        Step 1
+        ##########
```

Comment:
instead of multiline strings for these, should we use comments?  Saves some vertical space. They could be like:

```python
########### Step 1 #############
self.print_step(1, "....")
# ...

########### Step 2 #############
```

if you want to grep fast